scripts/iso_name_tagging.py

```diff
@@ -1,17 +1,30 @@
 #!/usr/bin/env python
 
-
 """Add intersecting feature(s) into a SAM file as a tag.
 
-Add intersecting feature names from a BED file as a tag to alignments in a SAM
-file using the format: feat-name|5p-shift|3p-shift|CIGAR|MD. Multiple 
-intersecting feature names are separated by a semi-colon. The 5p-shift and the
-3p-shift are calculated as the difference between the alignment and the feature
-start/end positions. If --extension is provided, it adjusts the feature start
-position by adding the given value and subtracts it from the end position. The
---id string specifies the feature name to be used within the attributes column
-in the BED file. If either the BED or the SAM file is empty, only the SAM file
-header will be returned.
+Build new names for the intersecting features from a BED file and add them as a
+tag to alignments in a SAM file using the format
+FEATURE_ID|5p-shift|3p-shift|CIGAR|MD. If either the BED or the SAM file is
+empty, only the SAM file header is returned.
+
+EXPECTED INPUT FILES
+The BED file must be the output a bedtools intersect call with -a being a GFF3
+file and -b a BAM file. If the GFF3 used in the bedtools intersect call has the
+features start and end coordinates extended, the number of additional
+nucleotides can be specified using the CLI option `--extension`. The SAM file
+must contian only the reads that have an intersecting feature.
+
+NAME CREATION and TAG ADDITION
+For each alignment, the name of the the intersecting feature will follow the
+format FEATURE_ID|5p-shift|3p-shift|CIGAR|MD. The CLI option `--id` specifies
+the feature identifier to be used as FEATURE_ID from within the attributes
+column in the BED file. The 5p-shift and the 3-p shift values are the
+difference between the feature start and end coordinates and the alignment
+start and end coordinates. If `--extension` is provided, the feature start
+position by are adjusted by adding the given value and subtracting it from the
+end position. If both, the 5p-shift and the 3p-shift, are within the range 
++/- extension + 1 the feature name is added to the alignment as the new tag
+"YW". Multiple intersecting feature names are separated by a semi-colon.
```

Comment:
You _absolutely_ need to give an example involving a BED record, a SAM record, the construted name and the new SAM record with the tag. And make sure the example covers all the special stuff, like extension etc. Or have multiple examples.